src/fromager/overrides.py

```diff
@@ -40,6 +41,23 @@ def _die_on_plugin_load_failure(
     raise RuntimeError(f"failed to load overrides for {ep.name}") from err
 
 
+def invoke(distname: str, method: str, default_fn: typing.Callable, **kwargs):
+    fn = find_override_method(distname, method)
+    if not fn:
+        fn = default_fn
+
+    sig = inspect.signature(fn)
+    # shallow copy because we do not want copies of dicts or objects passsed by the user as values to named args
+    fn_kwargs = kwargs.copy()
+    for arg_name in kwargs.keys():
```

Comment:
- `**kwargs` is already a shallow copy. You can safely modify the dict without affecting the caller
- `keys()` is bad style, `for key in kwargs` works.

```suggestion
    for arg_name in kwargs:
```

Reply:
Yes that's not why I was using the shallow copy. The issue was popping the keys while iterating

Reply:
ah, right!
The solution for that is `for arg_name in list(kwargs)`. This creates a copy of keys. The `dict.keys()` method returns a view (iterator), not a copy of keys.

Reply:
Ah nice thank you